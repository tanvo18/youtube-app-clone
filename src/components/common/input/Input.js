import React from 'react';
import InputStyled from './InputStyled';
import { calRem } from '../../../themes/mixins';
import PropTypes from 'prop-types';

const Input = (props) => {
  const { 
    inputType,
    inputClass,
    placeHolder,
    inputRef,
    handleKeyDown,
    borderRadius,
    inputWidth,
    inputHeight,
    inputBorder,
    inputPadding
  } = props;

  return (
    <InputStyled
      type={inputType}
      className={inputClass}
      placeholder={placeHolder}
      innerRef={inputRef}
      onKeyDown={handleKeyDown}
      borderRadius={borderRadius}
      width={inputWidth}
      height={inputHeight}
      inputBorder={inputBorder}
      inputPadding={inputPadding}
    >
    </InputStyled>
  );
};

Input.defaultProps = {
  type: 'text',
  width:`${calRem(200)}`,
  height:`${calRem(20)}`,
  inputBorder: `solid ${calRem(1)} #ccd3d9`,
  inputPadding: `${calRem(5)}`
};

Input.propType = {
  type: PropTypes.string.isRequired,
  className: PropTypes.string,
  placeHolder: PropTypes.string,
  borderRadius: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  inputBorder: PropTypes.string,
  inputPadding: PropTypes.string,
  onKeyDown: PropTypes.func.isRequired
};

export default Input;