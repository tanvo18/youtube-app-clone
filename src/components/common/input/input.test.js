import React from 'react';
import Input from './Input';
import renderer from 'react-test-renderer';

const props = {
  inputType: 'text',
  inputClass: 'test-input',
  placeHolder: 'Write something',
  borderRadius: '4px',
  inputWidth: '100px',
  inputHeight: '20px',
  inputBorder: 'solid 1px #000',
  inputPadding: '10px'
};

describe('test input', () => {
  const handleKeyDown = jest.fn();
  it('should render an input', () => {
    const wrapper = shallow(<Input handleKeyDown={handleKeyDown} {...props} />);
    expect(wrapper.exists()).toBe(true);
  });

  it('should call function handle click', () => {
    const wrapper = shallow(<Input handleKeyDown={handleKeyDown} {...props} />);
    wrapper.simulate('keyDown', {keyCode: 13});
    expect(handleKeyDown.mock.calls.length).toBe(1);
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<Input handleKeyDown={handleKeyDown} {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});