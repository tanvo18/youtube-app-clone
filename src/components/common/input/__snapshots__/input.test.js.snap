// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test input should render correctly 1`] = `
<input
  className="test-input sc-bdVaJa ktmTFK"
  height="20px"
  onKeyDown={[Function]}
  placeholder="Write something"
  type="text"
  width="100px"
/>
`;
